drivers/health_thrive_assessment/app/controllers/health_thrive_assessment/assessments_controller.rb

```diff
@@ -0,0 +1,77 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HealthThriveAssessment
+  class AssessmentsController < HealthController
+    include AjaxModalRails::Controller
+
+    before_action :set_patient
+    before_action :set_assessment, only: [:edit, :update, :show, :destroy]
+
+    def new
+      if @patient.thrive_assessments.in_progress.exists?
+        @assessment = @patient.thrive_assessments.in_progress.first
+      else
+        @assessment = @patient.thrive_assessments.create!(user: current_user)
+      end
+      redirect_to edit_client_health_thrive_assessment_assessment_path(@client, @assessment)
+    end
+
+    def edit
+    end
+
+    def update
+      @assessment.update(assessment_params)
+      redirect_to client_health_careplans_path(@client)
```

Comment:
Should this be `resond_with`?  It tends to handle errors more gracefully.  This doesn't seem to care if the update is unable to save...